cmd/operator-sdk/bundle/validate.go

```diff
@@ -188,23 +216,39 @@ func validateBundleContent(logger *log.Entry, manifestsDir string) ([]apierrors.
 	if err != nil {
 		return nil, err
 	}
-
 	return internalregistry.ValidateBundleContent(logger, bundle, mediaType), nil
 }
 
 // checkResults logs warnings and errors in results, and returns true if at
 // least one error was encountered.
-func checkResults(results []apierrors.ManifestResult) (hasErrors bool) {
+func checkResults(results []apierrors.ManifestResult, res *output.Result) {
+	hasErrors := false
 	for _, r := range results {
 		for _, w := range r.Warnings {
-			log.Warnf("%s validation: [%s] %s", r.Name, w.Type, w.Detail)
+			res.AddWarn(w)
 		}
 		for _, e := range r.Errors {
-			log.Errorf("%s validation: [%s] %s", r.Name, e.Type, e.Detail)
-		}
-		if r.HasError() {
+			res.AddError(e)
 			hasErrors = true
 		}
+
+	}
+	if hasErrors {
+		res.AddError(errors.New("invalid bundle content"))
```

Comment:
This error (and checking `hasErrors` in general here) is extraneous since we already have errors from the caller.